code/game/objects/items/weapons/tools/upgrades.dm

```diff
@@ -0,0 +1,445 @@
+	desc = "A small visible laser which can be strapped onto any tool, giving an accurate representation of its target. Helps improve precision"
+	icon_state = "laser_guide"
+	prefix = "laser-guided"
+	precision = 10
+	matter = list(MATERIAL_PLASTIC = 1, MATERIAL_URANIUM = 1)
+
+
+//Fits onto generally small tools that require precision, especially surgical tools
+//Doesn't work onlarger things like crowbars and drills
+/obj/item/weapon/tool_upgrade/refinement/stabilized_grip
+	name = "gyrostabilized grip"
+	desc = "A fancy mechanical grip that partially floats around a tool, absorbing tremors and shocks. Allows precise work with a shaky hand"
+	icon_state = "stabilizing"
+	required_qualities = list(QUALITY_CUTTING,QUALITY_WIRE_CUTTING, QUALITY_SCREW_DRIVING, QUALITY_WELDING,
+	QUALITY_PULSING, QUALITY_CLAMPING, QUALITY_CAUTERIZING, QUALITY_BONE_SETTING, QUALITY_LASER_CUTTING)
+	prefix = "stabilized"
+	precision = 10
+
+/obj/item/weapon/tool_upgrade/refinement/magbit
+	name = "magnetic bit"
+	desc = "Magnetises tools used for handling small objects, reducing instances of dropping screws and bolts."
+	icon_state = "magnetic"
+	required_qualities = list(QUALITY_SCREW_DRIVING, QUALITY_BOLT_TURNING, QUALITY_CLAMPING, QUALITY_BONE_SETTING)
+	prefix = "magnetic"
+	precision = 10
+
+
+/obj/item/weapon/tool_upgrade/refinement/ported_barrel
+	name = "ported barrel"
+	desc = "A barrel extension for a welding tool which helps manage gas pressure and keep the torch steady."
+	icon_state = "ported_barrel"
+	required_qualities = list(QUALITY_WELDING)
+	prefix = "ported"
+	precision = 12
+	bulk_mod = 1
+
+
+
+
+
+
+
+
+
+// 		AUGMENTS: MISCELLANEOUS AND UTILITY
+//------------------------------------------------
+
+//Allows the tool to use a cell one size category larger than it currently uses. Small to medium, medium to large, etc
+/obj/item/weapon/tool_upgrade/augment/cell_mount
+	name = "heavy cell mount"
+	icon_state = "cell_mount"
+	desc = "A bulky adapter which allows oversized power cells to be installed into small tools"
+	req_cell = TRUE
+	prefix = "medium-cell"
+	bulk_mod = 1
+
+/obj/item/weapon/tool_upgrade/augment/cell_mount/can_apply(var/obj/item/weapon/tool/T, var/mob/user)
+	.=..()
+	if (.)
+		if (T.suitable_cell == /obj/item/weapon/cell/medium || T.suitable_cell == /obj/item/weapon/cell/small)
+			if (T.cell)
+				user << SPAN_DANGER("You'll need to remove the power cell before installing this upgrade. It won't be compatible afterwards")
+				return FALSE
+			return TRUE
+		else
+			return FALSE
+
+/obj/item/weapon/tool_upgrade/augment/cell_mount/apply_values()
+	if (!holder)
+		return
+	if (holder.suitable_cell == /obj/item/weapon/cell/medium)
+		holder.suitable_cell = /obj/item/weapon/cell/large
+		prefix = "large-cell"
+	else if (holder.suitable_cell == /obj/item/weapon/cell/small)
+		holder.suitable_cell = /obj/item/weapon/cell/medium
+		prefix = "medium-cell"
+	..()
+
+
+
+
+//Stores moar fuel!
+/obj/item/weapon/tool_upgrade/augment/fuel_tank
```

Comment:
This upgrade needs more degradation multiplier, to make tool with this installed less reliable.